src/dsp_tools/utils/excel_to_json/properties.py

```diff
@@ -0,0 +1,457 @@
+        prop_row: row from a pd.DataFrame that defines a property
+        row_count: row number of Excel file
+        excel_filename: name of the original Excel file
+
+    Returns:
+        dict object of the property
+
+    Raises:
+        UserError if there are any formal mistakes in the "gui_attributes" column
+    """
+    _property = {x: prop_row[x] for x in mandatory_properties}
+    # These are also mandatory but require formatting
+    _property.update(
+        {"labels": utl.get_labels(df_row=prop_row), "super": [s.strip() for s in prop_row["super"].split(",")]}
+    )
+    non_mandatory = {
+        "comments": utl.get_comments(df_row=prop_row),
+        "gui_attributes": _get_gui_attribute(df_row=prop_row, row_num=row_count, excel_filename=excel_filename),
+    }
+    # These functions may return None, this is checked before the update
+    _property = utl.update_dict_ifnot_value_none(additional_dict=non_mandatory, to_update_dict=_property)
+    return _property
+
+
+def _check_gui_attributes(check_df: pd.DataFrame) -> dict[str, pd.Series] | None:
+    """
+    This function takes a pd.DataFrame and checks if the "gui_attributes" column is filled correctly. If any
+    or all of the checks fail, it creates a dictionary with a pd.Series as value which contains True for all rows where
+    there is a problem otherwise, it returns None.
+
+    Args:
+        check_df: pd.DataFrame that should be checked
+
+    Returns:
+        A dictionary with a pd.Series that contains the information where there is a problem or None if all the
+        checks passed.
+
+    Raises:
+        UserError if any of the checks fail
+    """
+    mandatory_attributes = ["Spinbox", "List"]
+    mandatory_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=mandatory_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=True,
+    )
+    no_attributes = ["Checkbox", "Date", "Geonames", "Richtext", "TimeStamp"]
+    no_attribute_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=no_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=False,
+    )
+    # If neither has a problem, we return None
+    if mandatory_check is None and no_attribute_check is None:
+        return None
+    # If both have problems, we combine the series
+    elif mandatory_check is not None and no_attribute_check is not None:
+        final_series = pd.Series(np.logical_or(mandatory_check, no_attribute_check))
+    elif mandatory_check is not None:
+        final_series = mandatory_check
+    else:
+        final_series = no_attribute_check
+    # The boolean series is returned
+    return {"wrong gui_attributes": final_series}
+
+
+def _check_missing_values_in_row_raise_error(to_check_df: pd.DataFrame, excel_filename: str) -> None:
```

Comment:
What is it here that is unclear?

Reply:
"Clean code reads like well-written prose." (Robert C. Martin)

If we look at it from the caller side: When we use a function, we want to know what it does. Some examples from my suggestions in other comments:

| your name                            | my suggestion                               |
| ------------------------------------ | ------------------------------------------- |
| check_duplicate_raise_error()        | check_column_for_duplicates()               |
| _search_validation_error()           | _prepare_err_msg_from_json_validation_error |
| check_required_columns_raise_error() | check_that_df_contains_required_columns()   |

The reason I prefer my suggestions is that it's almost like telling the computer: "do that for me". So the reason I struggle with your name `_check_missing_values_in_row_raise_error()` is that I cannot translate it into a spoken command like "Hey computer, please check the missing values in the row and raise an error".

See also: https://dev.to/pksilen/27-tips-for-making-your-code-read-like-beautifully-written-prose-4aab

Reply:
Tip #6: Use a preposition in the function name only when necessary: You don’t need to add a preposition to a function name if the preposition can be assumed (i.e., the preposition is implicit).
?

Reply:
What is your question?

Reply:
Would this not violate this rule?

Reply:
Do you mean that Tip no. 6 from dev.to violates Robert Martin's quote? These rules are not natural laws, rather they point you in a certain direction. So in the present case (discussing about `_check_missing_values_in_row_raise_error`), it's too far away from natural language. Especially, I struggle with the `_raise_error` part. If you delete that, it would already be better.